### PR TITLE
Implement Structured Data Feature Specs

### PR DESCRIPTION
**Pull request by matteeyao**
Developer Story
---
**As a** developer
**I want** to refactor my codebase
**So that** the codebase is easier to follow

Acceptance Criteria
---
**PR**
**Given** I complete this set of requirements
**Before** moving onto the next set of features
**Submit** a PR

**Fully Tested**
**Given** I am a developer
**When** I write the functionality and code for these specifications
**Then** they should be fully and well tested

### CODE REVIEW

File: handler_test.go [hunk 1]
```diff
@@ -240,3 +240,109 @@ func TestEchoBodyEndpoint(t *testing.T) {
...
+	if actualBody := resp.Body; strings.Compare(actualBody, expectedBody) != 0 {
+		t.Errorf("expected %s, got %s", expectedBody, actualBody)
+	}
+}
+
+func TestHTMLResponseEndpoint(t *testing.T) {
+	req, resp := &req, new(Response)
+
+	expectedBody := "<html><body><p>HTML Response</p></body></html>"
+	req.Body = expectedBody
```
Comment (TakaGoto) [inline handler_test.go:272]: does the request need to have the body here? or is this a test for echo requests?

File: handler_test.go [hunk 2]
```diff
@@ -240,3 +240,109 @@ func TestEchoBodyEndpoint(t *testing.T) {
...
+	if actualBody := resp.Body; strings.Compare(actualBody, expectedBody) != 0 {
+		t.Errorf("expected %s, got %s", expectedBody, actualBody)
+	}
+}
+
+func TestJSONResponseEndpoint(t *testing.T) {
+	req, resp := &req, new(Response)
+
+	expectedBody := "{\"key1\":\"value1\",\"key2\":\"value2\"}"
+	req.Body = expectedBody
```
Comment (TakaGoto) [inline handler_test.go:296]: same question as above

File: handler_test.go [hunk 3]
```diff
@@ -240,3 +240,109 @@ func TestEchoBodyEndpoint(t *testing.T) {
...
+	if actualBody := resp.Body; strings.Compare(actualBody, expectedBody) != 0 {
+		t.Errorf("expected %s, got %s", expectedBody, actualBody)
+	}
+}
+
+func TestXMLResponseEndpoint(t *testing.T) {
+	req, resp := &req, new(Response)
+
+	expectedBody := "<note><body>XML Response</body></note>"
+	req.Body = expectedBody
```
Comment (TakaGoto) [inline handler_test.go:325]: same q as above

File: handler.go [hunk 4]
```diff
@@ -75,3 +75,34 @@ func EchoBody (w *Response, r *Request) {
...
+	w.Header.Add("Content-Type", "text/html;charset=utf-8")
+	expectedBody := "<html><body><p>HTML Response</p></body></html>"
+	w.Body = expectedBody
+}
+
+func JSONResponse(w *Response, r *Request) {
+	ApplyStatusToResponse(w, StatusOK)
+	w.Header.Del("Content-Type")
+	w.Header.Add("Content-Type", "application/json;charset=utf-8")
+	expectedBody := "{\"key1\":\"value1\",\"key2\":\"value2\"}"
```
Comment (TakaGoto) [inline handler.go:96]: is the body supposed to be hard coded here? or is it supposed to read the data coming from the request?

File: handler.go [hunk 5]
```diff
@@ -75,3 +75,34 @@ func EchoBody (w *Response, r *Request) {
...
+	w.Body = expectedBody
+}
+
+func XMLResponse(w *Response, r *Request) {
+	ApplyStatusToResponse(w, StatusOK)
+	w.Header.Del("Content-Type")
+	w.Header.Add("Content-Type", "application/xml;charset=utf-8")
+	expectedBody := "<note><body>XML Response</body></note>"
```
Comment (TakaGoto) [inline handler.go:104]: is the body supposed to be hard coded here? or is it supposed to read the data coming from the request?

File: handler.go [hunk 6]
```diff
@@ -75,3 +75,34 @@ func EchoBody (w *Response, r *Request) {
...
+	ApplyStatusToResponse(w, StatusOK)
+	w.Header.Add("Content-Type", "text/plain;charset=utf-8")
+	expectedBody := "text response"
+	w.Body = expectedBody
+}
+
+func HTMLResponse(w *Response, r *Request) {
+	ApplyStatusToResponse(w, StatusOK)
+	w.Header.Add("Content-Type", "text/html;charset=utf-8")
+	expectedBody := "<html><body><p>HTML Response</p></body></html>"
```
Comment (TakaGoto) [inline handler.go:88]: is the body supposed to be hard coded here? or is it supposed to read the data coming from the request?

File: handler_test.go (hunk 1 above)
Comment (matteeyao) [reply handler_test.go:272]: ✅ I misread the specs here, request does not need body here and it's not supposed to echo a request here. Changed accordingly.

File: handler_test.go (hunk 2 above)
Comment (matteeyao) [reply handler_test.go:296]: ✅

File: handler_test.go (hunk 3 above)
Comment (matteeyao) [reply handler_test.go:325]: ✅

File: handler.go (hunk 4 above)
Comment (matteeyao) [reply handler.go:96]: The body is what was required by the integration specs, but there is no data sent from the request.

File: handler.go (hunk 5 above)
Comment (matteeyao) [reply handler.go:104]: The body is what was required by the integration specs, but there is no data sent from the request.

File: handler.go (hunk 6 above)
Comment (matteeyao) [reply handler.go:88]: The body is what was required by the integration specs, but there is no data sent from the request.

**Pull request merged**